metadata-ingestion/src/datahub/ingestion/source/tableau/tableau.py

```diff
@@ -2428,10 +2428,13 @@ def parse_custom_sql(
             ]
         ],
     ) -> Optional["SqlParsingResult"]:
-        database_info = datasource.get(c.DATABASE) or {
-            c.NAME: c.UNKNOWN.lower(),
-            c.CONNECTION_TYPE: datasource.get(c.CONNECTION_TYPE),
+        database_info = {
+            c.ID: datasource.get(c.DATABASE, {}).get(c.ID),
```

Comment:
the field `database` is Optional in [GraphQL Spec](https://help.tableau.com/current/api/metadata_api/en-us/reference/customsqltable.doc.html), i.e. the key database can have value set as None. This case does not get covered by the statement `datasource.get(c.DATABASE, {}).get(c.ID)` which would throw error. Maybe replace it by `(datasource.get(c.DATABASE) or {}).get(c.ID)` ?

I'd also suggest a refactor to avoid this clause repeated over 3 times, if that makes sense.